pkg/file/builder.go

```diff
@@ -175,6 +177,53 @@ func (b *stateBuilder) consumerGroups() {
 		return
 	}
 
+	// Load all existing consumer groups in to the intermediate state for
+	// foreign key lookups if we're doing a partial apply
+	if b.isPartialApply {
+		consumerGroups, err := b.currentState.ConsumerGroups.GetAll()
+		if err != nil {
+			b.err = err
+			return
+		}
+		for _, cg := range consumerGroups {
+			// Add to intermediate state for lookups
+			err = b.intermediate.ConsumerGroups.Add(*cg)
+			if err != nil {
+				b.err = err
+				return
+			}
+
+			// Also add to rawState to perform a lookup later
+			rawCg := &kong.ConsumerGroupObject{
+				ConsumerGroup: &kong.ConsumerGroup{
+					ID:        cg.ID,
+					Name:      cg.Name,
+					CreatedAt: cg.CreatedAt,
+					Tags:      cg.Tags,
+				},
+			}
+
+			// Fetch consumers for this CG
+			consumerGroupConsumers, err := b.currentState.ConsumerGroupConsumers.GetAll()
+			if err != nil {
+				b.err = err
+				return
+			}
+
+			var consumers []*kong.Consumer
+			for _, cgc := range consumerGroupConsumers {
+				if cgc.ConsumerGroup.ID == cg.ID {
+					consumers = append(consumers, cgc.Consumer)
+				}
+			}
+
+			rawCg.Consumers = consumers
```

Comment:
Move `rawCg` close to where it's used

```suggestion
			// Fetch consumers for this CG
			consumerGroupConsumers, err := b.currentState.ConsumerGroupConsumers.GetAll()
			if err != nil {
				b.err = err
				return
			}

			var consumers []*kong.Consumer
			for _, cgc := range consumerGroupConsumers {
				if cgc.ConsumerGroup.ID == cg.ID {
					consumers = append(consumers, cgc.Consumer)
				}
			}

			// Also add to rawState to perform a lookup later
			rawCg := &kong.ConsumerGroupObject{
				ConsumerGroup: &kong.ConsumerGroup{
					ID:        cg.ID,
					Name:      cg.Name,
					CreatedAt: cg.CreatedAt,
					Tags:      cg.Tags,
				},
				Consumers: consumers,
			}
```